### PR TITLE
add notes about TR+match in docs

### PR DESCRIPTION
**Pull request by pnwamk**
Adds some advice regarding using Typed Racket and match.

See https://github.com/racket/typed-racket/pull/759.

### CODE REVIEW

File: typed-racket-doc/typed-racket/scribblings/guide/caveats.scrbl [hunk 1]
```diff
@@ -180,3 +180,66 @@ circumvent the protections of the type system.
 
 Contract boundaries installed for typed-untyped interaction may cause
 significant slowdowns. See @secref{contract-costs} for details.
+
+@section{Pattern Matching and Occurrence Typing}
+
+
+Because Typed Racket type checks code @emph{after} macro
+expansion, certain forms---such as @racket[match]---are
+difficult for Typed Racket to completely reasoning about. In
```
Comment (greghendershott) [inline typed-racket-doc/typed-racket/scribblings/guide/caveats.scrbl]: "reasoning" => "reason"

Comment (greghendershott) [reply typed-racket-doc/typed-racket/scribblings/guide/caveats.scrbl]: If you're picky about split infinitives (I'm not) you could say "to reason about completely" or "to understand fully" or whatever.

Comment (pnwamk) [reply typed-racket-doc/typed-racket/scribblings/guide/caveats.scrbl]: Thanks

**Comment by pnwamk** [comment]
@samth did you want any additional improvements/tweaks? I should I merge this?

**Comment by samth** [comment]
This looks good to me.

**Pull request merged**